pkg/models/activity.go

```diff
@@ -42,3 +60,34 @@ func NewActivity(actorID uuid.UUID, entityID uuid.UUID, activityType ActivityTyp
 		ActivityType: activityType,
 	}
 }
+
+// ParseRawActivityMetaData conditionally parses the raw meta data JSON into the appropriate struct based on ActivityType
+// It returns a boolean indicating if parsing was successful and an error if any occurred during parsing
+func (a *Activity) ParseRawActivityMetaData() (bool, error) {
```

Comment:
I'm not quite seeing why we need a bool and error return type here, anytime there is an error we return false, and vice versa, so they seem coupled together so that your only returns are 

(false, err) or (true, nil)

Reply:
Yeah, I think that's fair! I'll work to adjust it